test/utils/testutils.fpp

```diff
@@ -0,0 +1,347 @@
+!
+! Distributed Linear Algebra with Future (DLAF)
+!
+! Copyright (c) 2018-2024, ETH Zurich
+! All rights reserved.
+!
+! Please, refer to the LICENSE file in the root directory.
+! SPDX-License-Identifier: BSD-3-Clause
+!
+
+#:set precision = ['sp', 'dp']
+#:set types = ['real', 'complex']
+#:set symbols = {('sp', 'real'): 's', ('sp', 'complex'): 'c', ('dp', 'real'): 'd', ('dp', 'complex'): 'z'}
+module testutils
+
+#ifdef DLAF_FORTRAN_WITH_MPI_F08
+   use mpi_f08
+#else
+   use mpi
+#endif
```

Comment:
Ah. Well spotted. Can you please share what the problem/complaint was? I haven't seen it, or I completely missed it.

The F08 version is unfortunately not tested in CI, because the `mpich` version we have to use there is too old. =(

Reply:
```bash
ialberto@daint101:~/workspace/dlaf-fortran/build> make
[  5%] Built target DLAF_Fortran
[  7%] Generating testutils.f90
[ 10%] Building Fortran object test/utils/CMakeFiles/testutils.dir/testutils.f90.o
/users/ialberto/workspace/dlaf-fortran/build/test/utils/testutils.f90:13:2:

   13 | #ifdef DLAF_FORTRAN_WITH_MPI_F08
      |  1
Warning: Illegal preprocessor directive
/users/ialberto/workspace/dlaf-fortran/build/test/utils/testutils.f90:15:2:

   15 | #else
      |  1
Warning: Illegal preprocessor directive
/users/ialberto/workspace/dlaf-fortran/build/test/utils/testutils.f90:17:2:

   17 | #endif
      |  1
Warning: Illegal preprocessor directive
/users/ialberto/workspace/dlaf-fortran/build/test/utils/testutils.f90:14:8:

   14 |    use mpi_f08
      |        1
Fatal Error: Cannot open module file 'mpi_f08.mod' for reading at (1): No such file or directory
compilation terminated.
make[2]: *** [test/utils/CMakeFiles/testutils.dir/build.make:79: test/utils/CMakeFiles/testutils.dir/testutils.f90.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:595: test/utils/CMakeFiles/testutils.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

Reply:
> When using the [Ninja](https://cmake.org/cmake/help/latest/generator/Ninja.html#generator:Ninja) generator, all source files are first preprocessed in order to generate module dependency information.

The issue is reproducible with `make`.

Reply:
Since the preprocessor directives are localised in a single file, I enabled the `Fortran_PREPROCESS` property on the associated target only. We can eventually re-visit this if needed. Thank you for spotting this problem!